### ISSUE TITLE
inner join

### ISSUE THREAD

**Issue by jimrothstein** [LucyNjoki/R-LadiesAbuja_Data-Manipulation-with-data.table-in-R#3]
now:   Inner Join, done with `merge` and `y[x`

If you carefully at results, rows are the same, but the `column` order differs.
Maybe just not important, but I need to review to figure it out.
(if these little things are not useful, just tell me, I'll stop.    However, the effort is useful to me.)

merge(x = x, y = y, by = "id")
```
   **id  x  y**
1:  1 45  8
2:  4 90 19
3:  6  8  2
4:  6  8  4

- Let's do an inner join the two data.tables now using `data.table` syntax

```{r inner_join_data.table_syntax}
y[x, on = .(id), nomatch = 0]

```

  ** id  y  x**
1:  1  8 45
2:  4 19 90
3:  6  2  8
4:  6  4  8
>

**Comment by jimrothstein** [comment]
This stack overflow post seems related:  https://stackoverflow.com/questions/12773822/why-does-xy-join-of-data-tables-not-allow-a-full-outer-join-or-a-left-join
@mnel  takes note of column switch and fixes it with `setcolorder()`

**Comment by LucyNjoki** [comment]
Indeed, yes. I am just noticing that now. Thank you! Please, go ahead, these tiny details are **very helpful**! :)  I'll read the link shared more carefully and amend my code asap.
